server/src/infra/repositories/filesystem.provider.ts

```diff
@@ -132,5 +133,13 @@ export class FilesystemProvider implements IStorageRepository {
     });
   }
 
+  async watch(paths: string | ReadonlyArray<string>, options?: WatchOptions): Promise<FSWatcher> {
+    // Stop any previous watchers of this library
+
+    const watcher = chokidar.watch(paths, options);
+
+    return watcher;
+  }
+
```

Comment:
There's not a lot of point is "externalizing" this to a repository if you use the same input, options, and return value. It is essentially the same as calling it directly. It would be better to have an abstract interface for "watching" that was not chokidir specific.

Reply:
Right, before my last refactor the repo contained more logic. After moving most of that back to the library service it could be easiest just to return the chokidar call itself. Your call, this was done to harmonize with other things like the crawler.